virtualizarr/writers/icechunk.py

```diff
@@ -1,11 +1,14 @@
-from typing import TYPE_CHECKING, cast
+from typing import TYPE_CHECKING, Optional, cast
 
 import numpy as np
+import zarr  # type: ignore[import-untyped]
 from xarray import Dataset
 from xarray.backends.zarr import encode_zarr_attr_value
 from xarray.core.variable import Variable
+from zarr import Array
```

Comment:
Actually sorry this and the other zarr import should just be removed. You might need to move `from zarr import Array` inside `if TYPE_CHECKING`.